scripts/execwhacker.py

```diff
@@ -51,9 +53,71 @@ class ProcessSource(Enum):
 processes_killed = Counter("cryptnono_execwhacker_processes_killed_total", "Total number of processes killed", ["source"])
 
 
-def kill_if_needed(banned_strings_automaton, allowed_patterns, cmdline, pid, source):
+def log_and_kill(pid, cmdline, b, source, lookup_container):
+    """
+    Attempt to lookup the container details for a given PID, then log and kill it
+
+    This makes an external blocking call to lookup the container details
+
+    Returns True if the process was killed, False if it was not found
+    """
+    # TODO: Should we switch to structured JSON logging instead?
```

Comment:
I've had a look at structured logging in https://github.com/manics/cryptnono/commit/f11ff79935a862df227514ad29524ef7f8774d2c
It makes it a lot easier to construct parsable log messages with multiple fields.